### ISSUE TITLE
Allow to increment a env var for instances of an application started

### ISSUE THREAD

**Issue by vmarchaud** [Unitech/pm2#2743]
For example i want to run my application in fork_mode with 4 process, but only one declaration.

```
{
    name       : "api-app",
    script     : "./api.js",
    instances  : 4,
    exec_mode  : "fork"
    env: {
      PORT: 3000
    }
  }
```

And add something like that : 
```
{
    name       : "api-app",
    script     : "./api.js",
    instances  : 4,
    increment_foreach: 'PORT',
    exec_mode  : "fork"
    env: {
      PORT: 3000
    }
  }
```

**Comment by vmarchaud** [comment]
Function responsible for this : https://github.com/Unitech/pm2/blob/1c3fcb83b65e430a5734d576610d27cc7a0bf4fd/lib/God.js#L355
Specially this loop : https://github.com/Unitech/pm2/blob/1c3fcb83b65e430a5734d576610d27cc7a0bf4fd/lib/God.js#L366

It should look like : 
```
    var instance_id = 0;
    var increment = 0;

    (function ex(i) {
      if (i <= 0) {
        if (cb) return cb(null, arr);
        return false;
      }

     // should look like that
     if (env.en && env.env.increment_foreach)
       env[env.env.increment_foreach] = increment++;

      env.NODE_APP_INSTANCE = instance_id++;
      env.vizion_running = false;

      if (env.env && env.env.vizion_running)
        env.env.vizion_running = false;

      return God.executeApp(Utility.clone(env), function(err, clu) {
        if (err) return ex(i - 1);
        arr.push(Utility.clone(clu));
        God.notify('start', clu, true);
        return ex(i - 1);
      });
    })(env.instances);
```

**Comment by btmdave** [comment]
Can this be merged in?

**Comment by vmarchaud** [comment]
It need tests cause like @soyuka said, it doesn't work like that, you need to put the option in the options schema inside `lib/API/schema.json`.
If you want this to be merged it will need some more work, otherwise i will make it myself when i will have to do so.

**Comment by btmdave** [comment]
@vmarchaud I've updated here: https://github.com/btmdave/pm2/commit/37b42b87016b78b560665e2e3c00ddf25494bd49

I changed to `increment` and I assume, it's increment that needs to be added to schema? I've added that.  Really not sure how to setup a test case for this though.

**Comment by vmarchaud** [comment]
Published under PM2 `2.5` : `npm install -g pm2@latest && pm2 update`
